gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -383,13 +383,32 @@ class {{ async_method_name_prefix }}{{ service.name }}RestInterceptor:
     {% endif %}
         """Post-rpc interceptor for {{ method.name|snake_case }}
 
-        Override in a subclass to manipulate the response
+        Deprecated. This interceptor is deprecated. Please use the `post_{{ method.name|snake_case }}_with_metadata`
+        interceptor instead. Override in a subclass to read or manipulate the response
         after it is returned by the {{ service.name }} server but before
-        it is returned to user code.
+        it is returned to user code. This `post_{{ method.name|snake_case }}` iterceptor runs
+        before the `post_{{ method.name|snake_case }}_with_metadata` iterceptor.
         """
         return response
 
+    {% if not method.server_streaming %}
+    {{ async_prefix }}def post_{{ method.name|snake_case }}_with_metadata(self, response: {{method.output.ident}}, {{ client_method_metadata_argument() }}) -> Tuple[{{method.output.ident}}, {{ client_method_metadata_type() }}]:
+    {% else %}
+    {{ async_prefix }}def post_{{ method.name|snake_case }}_with_metadata(self, response: rest_streaming{{ async_suffix }}.{{ async_method_name_prefix }}ResponseIterator, {{ client_method_metadata_argument() }}) -> Tuple[rest_streaming{{ async_suffix }}.{{ async_method_name_prefix }}ResponseIterator, {{ client_method_metadata_type() }}]:
     {% endif %}
+        """Post-rpc interceptor for {{ method.name|snake_case }}
+
+        Override in a subclass to read or manipulate the response or metadata after it
+        is returned by the {{ service.name }} server but before it is returned to user code.
+        This `post_{{ method.name|snake_case }}_with_metadata` interceptor runs after the
+        `post_{{ method.name|snake_case }}` iterceptor. If the response is modified in
+        `post_{{ method.name|snake_case }}`, the modified response will be used in
+        `post_{{ method.name|snake_case }}_with_metadata`. This `post_{{ method.name|snake_case }}_with_metadata`
+        interceptor is recommended for new development instead of `post_{{ method.name|snake_case }}`.
```

Comment:
```suggestion
        Override in a subclass to read or manipulate the response or metadata after it
        is returned by the {{ service.name }} server but before it is returned to user code.
        
        We recommend only using this `post_{{ method.name|snake_case }}_with_metadata`
        interceptor in new development instead of the `post_{{ method.name|snake_case }}` interceptor.

        When both interceptors are used, this `post_{{ method.name|snake_case }}_with_metadata` interceptor runs after the
        `post_{{ method.name|snake_case }}` interceptor. The (possibly modified) response returned by 
        `post_{{ method.name|snake_case }}` will be passed to
        `post_{{ method.name|snake_case }}_with_metadata`. 
        
```

Reply:
Done in https://github.com/googleapis/gapic-generator-python/pull/2299/commits/11b803f8642d02ad0c0ed8f641b91209d812b170